Veeam App for Microsoft Sentinel

nikita-konoplin opened the pull request:
Required items, please complete
   
   Change(s):
   - Initial version of Veeam App for Microsoft Sentinel

   Reason for Change(s):
   - Initial version

   Version Updated:
   - Initial version

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - No

nikita-konoplin commented:
@microsoft-github-policy-service agree company="Veeam Software"

v-shukore commented:
Hi @nikita-konoplin,

I hope you're doing well.

It looks like the folder structure within the Data Connector may need some adjustments. Could you please review it and make the necessary corrections? Also, kindly update the .zip file name to reflect the corrected folder structure. For reference, you may look at the "Cloudflare" solution available in the GitHub repository.
Additionally, could you please share the invocation logs of the running data connector and provide some sample data for reference?

Please also correct the indentation in the Data Input file to ensure all content is properly aligned.

It appears that the playbook images are missing from the Playbook folder. Kindly add images that demonstrate the playbook in action. You can refer to the structure used in the "CiscoUmbrella" solution in the GitHub repository.

To resolve the KQL validation issue, please ensure that the required column is included in the query. Also, in the Analytic Rule folder, there is a PowerShell script file named "update_descriptions"—kindly remove this file from Analytic Rule folder.

Regarding the logo validation issue, please open the logo file in Visual Studio Code, locate the id property, and replace it with a new GUID (a 16-character unique identifier). You can easily generate one online.

Thank you so much!

nikita-konoplin commented:
Hi @v-shukore,
We’ve updated the solution structure and added the required information. Could you please take a look and rerun the validations?

It’s not clear to us where we can obtain the "invocation logs of the running data connector" and where they should be attached. Could you please clarify?

Thank you,
Nikita

v-shukore commented:
Hi @nikita-konoplin, 
Could you please also resolve the KQL and logo validations? Additionally, it seems the playbook images are missing from the Playbook folder. Kindly add images that demonstrate the playbook is running. For reference, you may follow the structure used in the `CiscoUmbrella` solution from our repository.
I also noticed that the zip file name in the Data Connector folder appears to be incorrect—it should follow the naming convention like "VeemConn." Could you please update it accordingly?
Lastly, please include the requirements.txt, host, and proxy files in the Data Connector folder, similar to how they are included in the `Cloudflare` solution from our repository.
Thank you!

nikita-konoplin commented:
Hello @v-shukore ,
We have added screenshots to the images folder for each of our Playbooks inside the Solutions/Veeam/Playbooks/ folder, and renamed our .zip file to VeeamConn.zip.
Regarding having the same structure of the repo as Cloudflare, we cannot fully do that due to the fact that we're using C# for our application, not Python as Cloudflare does. So our folder structure is a bit different. You can find the Veeam.Sentinel.FunctionApp.csproj file (which is the analogue of requirements.txt for C#) and host.json in the Solutions/Veeam/Data Connectors/AzureFunctionVeeam/Veeam.Sentinel.FunctionApp/ folder.
We have updated the solutionmetadata file with the correct value.
Regarding the invocation logs:
Here's the screenshot for the GetAllMalwareEventsAsync function:
<img width="1036" height="354" alt="image" src="https://github.com/user-attachments/assets/51800ba2-5d0c-4ca8-a6f5-e25e4e72b17f" />

It looks like these invocation details are tied to the function, not to the whole function app. We have 16 functions in our function app — should we provide you with a screenshot for each function?

Thank you!

v-shukore commented:
Hi @nikita-konoplin,
Could you please update the workbook image names to follow the standard naming convention? This applies to both the images in the workbook folder and the entries in the workbookMetadata file. As shown in the example below, the image name should end with either "white" or "black", depending on the theme of image.
<img width="328" height="79" alt="image" src="https://github.com/user-attachments/assets/d7bcbaaa-f667-4a48-ac01-4e020d9cc433" />
Thanks!!

nikita-konoplin commented:
Hi @v-shukore,
Looks like validation ([Detection Template Schema Validations](https://github.com/Azure/Azure-Sentinel/actions/runs/17458716258/job/49579903649?pr=12709)) failed because of changes in the main branch.

Expected exception to be <null> because In template **CiscoUmbrella**RequestBlocklistedFileType.yaml there was an error while parsing: Required property 'relevantTechniques' not found in JSON.

Please take a look.

v-shukore commented:
Hi @nikita-konoplin, 
Would you like to make any changes to this PR, or shall we proceed with merging it into the master branch?
Thanks!

nikita-konoplin commented:
Hi @v-shukore ,
Here it is https://github.com/Azure/Azure-Sentinel/pull/12752

v-shukore commented:
Thank You @nikita-konoplin for raising a PR.
